freqtrade/freqtradebot.py

```diff
@@ -476,6 +476,10 @@ def handle_onexchange_order(self, trade: Trade):
             if not trade.is_open:
                 # Trade was just closed
                 trade.close_date = trade.date_last_filled_utc
+                strategy_safe_wrapper(
+                    self.strategy.order_filled, default_retval=None)(
+                    pair=trade.pair, trade=trade, order=order_obj,
+                    current_time=datetime.now(timezone.utc))
```

Comment:
I think if you happen to run into this case - then you'll trigger the callback twice / multiple times.

The above "refind" logic will trigger the callback for every (new) order it finds (essentially orders where you manually sold assets on exchange).

If that triggers a trade close - the callback will trigger again - though in that case, there's no "acual" new order fill you can attach - so it'll trigger twice for the very last order.